lib/Optimizer/Dialect/CC/CCOps.cpp

```diff
@@ -1143,16 +1151,43 @@ ParseResult cudaq::cc::IfOp::parse(OpAsmParser &parser,
       parser.parseRParen() ||
       parser.resolveOperand(cond, i1Type, result.operands))
     return failure();
+  SmallVector<OpAsmParser::Argument, 4> regionArgs;
+  if (succeeded(parser.parseOptionalLParen())) {
+    // Linear type arguments.
+    SmallVector<OpAsmParser::UnresolvedOperand, 4> linearOperands;
+    if (parser.parseAssignmentList(regionArgs, linearOperands) ||
+        parser.parseRParen())
+      return failure();
+    Type wireTy = quake::WireType::get(builder.getContext());
+    for (auto argOperand : llvm::zip(regionArgs, linearOperands)) {
+      std::get<0>(argOperand).type = wireTy;
+      if (parser.resolveOperand(std::get<1>(argOperand), wireTy,
+                                result.operands))
+        return failure();
+    }
+  }
   if (parser.parseOptionalArrowTypeList(result.types))
     return failure();
-  if (parser.parseRegion(*thenRegion, /*arguments=*/{}, /*argTypes=*/{}))
+  if (!regionArgs.empty()) {
+    // Check that the result.types is compatible with the regionArgs. To be
+    // compatible, it must have at least as many linear types as there are
+    // region arguments. (It can have more.)
+    std::int64_t numRegionArgs = regionArgs.size();
```

Comment:
I'm surprised this doesn't cause a warning with implicit casting unsigned to signed. 🤷‍♂️

Reply:
Fortunately, it doesn't. I think warnings like that come into play with truncation though, so assigning an `unsigned long` to an `int` would generate a warning.

Reply:
Some `unsigned long` elements cannot be mapped to `std::int_t` elements (i.e. 2^63). That being said, I'm not going to lose any sleep at night about us having >2^63 elements in `regionArgs`, so I'll mark this as resolved. 😆